packages/grafana-data/src/types/datasource.ts

```diff
@@ -351,6 +351,18 @@ abstract class DataSourceApi<
    * @alpha -- experimental
    */
   getDefaultQuery?(app: CoreApp): Partial<TQuery>;
+
+  /*
+   * Optionally, use this method to report information about queries
+   * @alpha -- experimental
+   */
+  queriesOnInitDashboard?(
```

Comment:
maybe, to avoid confussion about what this function should do, we can name it `trackQueries` also, for the description we can clarify when it's executed:

```
   * Optionally, use this method to track information about queries. Only executed once, during the dashboard initialization.
```